models/contacts.js

```diff
@@ -1,19 +1,117 @@
-// const fs = require('fs/promises')
+const fs = require("fs/promises");
+const { nanoid } = require("nanoid");
 
-const listContacts = async () => {}
+const getContacts = async () => {
+  const contacts = fs.readFile("models/contacts.json");
+  return JSON.parse(await contacts);
+};
 
-const getContactById = async (contactId) => {}
+const saveContact = async (data) =>
+  await fs.writeFile("models/contacts.json", JSON.stringify(data));
 
-const removeContact = async (contactId) => {}
 
-const addContact = async (body) => {}
 
-const updateContact = async (contactId, body) => {}
+const listContacts = async () => {
+  try {
+    const contacts = await getContacts();
+    return contacts;
+  } catch (error) {
+    console.log(error);
+    throw error; 
+  }
+};
+
+
+const getContactById = async (contactId) => {
+  try {
+    const contacts = await getContacts();
+
+    return contacts.find((data) => data.id === contactId);
+  } catch (error) {
+    console.log(error);
+    throw error; 
+  }
+};
+
+
+const removeContact = async (contactId) => {
+  try {
+    const contacts = await getContacts();
+    const targetContact = contacts.find((data) => data.id === contactId);
```

Comment:
`data` has no specific meaning it can be everything. `contact` would be better.

Reply:
done